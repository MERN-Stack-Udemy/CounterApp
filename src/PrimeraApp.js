// import React from 'react';
import PropTypes from 'prop-types';

const PrimeraApp = ({ saludo, edad, idioma }) => {
  
  return (
    <>
      <h1>Hola soy {saludo}</h1>
      <p>att. Net Flanders y tengo {edad} anios</p>
      <p>este articulo esta en el idioma {idioma}</p>
    </>
  );
};

PrimeraApp.propTypes = {
  saludo : PropTypes.string.isRequired, 
  edad : PropTypes.number.isRequired
}
PrimeraApp.defaultProps = {
  idioma: "ingles",
}

export default PrimeraApp;
