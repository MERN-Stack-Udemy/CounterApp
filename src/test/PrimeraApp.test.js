
import { shallow } from "enzyme";
import PrimeraApp from "../PrimeraApp";


describe('Pruebas en <PrimeraApp />', () => {
  
  // test('debe mostrar el mensaje "Hola soy German"', () => {
  //   const saludo = 'Hola soy German';
  //   const {getByText} = render( <PrimeraApp saludo="German" edad={23} /> );
  //   expect( getByText( saludo )).toBeInTheDocument;
  // })
  
  test('deve mostrar <PrimeraApp /> correctamente', () => {
    const saludo = 'Hola soy German';
    const edad = 23;
    const wrapper = shallow( <PrimeraApp saludo = { saludo } edad = { edad } /> );

    expect(wrapper).toMatchSnapshot();

  })

  test('deve mostrar el idioma del articulo', () => {
    const saludo = 'Hola soy German';
    const edad = 23;
    const idioma = "ES"
    const parrafoIdioma = `este articulo esta en el idioma ${idioma}`

    const wrapper  = shallow(
      <PrimeraApp 
          saludo= {saludo} 
          edad= {edad} 
          idioma={idioma} 
      />
    );
    const testpParrafo = wrapper.find('#idioma').text()
    // console.log(testpParrafo)
    expect( testpParrafo ).toBe ( parrafoIdioma)
  })
  
  
})
