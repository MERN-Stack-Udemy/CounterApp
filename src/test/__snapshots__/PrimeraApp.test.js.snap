// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <PrimeraApp /> deve mostrar <PrimeraApp /> correctamente 1`] = `
<Fragment>
  <h1>
    Hola soy 
    Hola soy German
    !!!
  </h1>
  <p>
    att. Net Flanders y tengo 
    23
     anios
  </p>
  <p
    id="idioma"
  >
    este articulo esta en el idioma 
    ingles
  </p>
</Fragment>
`;
